tests/app/test_jinja_filters.py

```diff
@@ -442,3 +443,60 @@ def test_other_config_visibility(
 
     other = OtherConfig(Mock(), answer_schema_textfield)
     assert other.open is expected_visibility
+
+
+@patch("app.jinja_filters.flask_babel.get_locale", Mock(return_value="en_GB"))
+def test_calculated_summary_question():
+    rows = map_summary_item_config(
+        group={
+            "blocks": [
+                {
+                    "id": "first-number-block",
+                    "link": "/questionnaire/first-number-block/?return_to=final-summary",
+                    "question": {
+                        "id": "first-number-question",
+                        "type": "General",
+                        "title": "First Number Question Title",
+                        "answers": [
+                            {
+                                "id": "first-number-answer",
+                                "label": "First answer label",
+                                "value": 1,
+                                "type": "currency",
+                                "currency": "GBP",
+                            }
+                        ],
+                    },
+                },
+                {
+                    "id": "second-number-block",
+                    "link": "/questionnaire/second-number-block/?return_to=final-summary",
+                    "question": {
+                        "id": "second-number-question",
+                        "type": "General",
+                        "title": "Second Number Question Title",
+                        "answers": [
+                            {
+                                "id": "second-number-answer",
+                                "label": "Second answer label",
+                                "value": 1,
+                                "type": "currency",
+                                "currency": "GBP",
+                            }
+                        ],
+                    },
+                },
+            ],
+        },
+        summary_type="CalculatedSummary",
+        answers_are_editable=True,
+        no_answer_provided="No answer Provided",
+        edit_link_text="Change",
+        edit_link_aria_label="Change your answer for",
+        calculated_question={
+            "title": "Grand total of previous values",
+            "id": "calculated-summary-question",
+            "answers": [{"id": "calculated-summary-answer", "value": "£2.00"}],
+        },
+    )
+    assert len(rows) == 3
```

Comment:
To assert that this is not 3 rows with Grand total, I would just add something like:
```py
assert len(rows) == 3
assert rows[0].rowTitle == "First Number Question Title"
assert rows[1].rowTitle == "Second Number Question Title"
assert rows[2].rowTitle == "Grand total of previous values"
```

I would update the name to something more specific, like `test_calculated_summary_total_is_not_duplicated` since it isn't test all of calculated summary.

Reply:
Extra assertions added, test renamed, thanks!

Reply:
I don't think we currently have a test for the rest of the generated context. Is there a reason not to go a step further and assert the full content of rows?

Reply:
Grand total is not added to question rows, it has its own rows. So before the change it would be something like 
```py
>>>  len(rows)
5
>>>  rows[0].rowTitle
>>>  "First Number Question Title"
>>>  rows[1].rowTitle
>>>  "Grand total of previous values"
>>>  rows[2].rowTitle
>>>  "Second Number Question Title"
>>>  rows[3].rowTitle
>>>  "Grand total of previous values"
>>>  rows[4].rowTitle
>>>  "Grand total of previous values"
```

Reply:
I understand that this tests the defect, but there is still a gap in the tests for the rest of the context e.g. the edit link that should appear for the answer rows but not the totals.

Reply:
So we do have a `TestCalculatedSummaryContext` but there are gaps in that, so that could be updated. I think other than that, we only have some functional tests. The jinja filters wraps a layer on top of that in its own format which should really be tested also.

Reply:
Should I add some assertions? Something like:
```py
    assert (
        rows[0].rowItems[0].actions[0].ariaLabel
        == "Change your answer for First answer label"
    )
    assert (
        rows[0].rowItems[0].actions[0].attributes["data-qa"]
        == "first-number-answer-edit"
    )
    assert rows[0].rowItems[0].actions[0].attributes["data-ga"] == "click"
    assert rows[0].rowItems[0].actions[0].attributes["data-ga-category"] == "Summary"
    assert rows[0].rowItems[0].actions[0].attributes["data-ga-action"] == "Edit click"
    assert rows[0].rowItems[0].actions[0].text == "Change"
    hasattr(rows[2].rowItems[0], 'actions')
    assert len(rows[2].rowItems[0].attributes) == 1
    assert rows[2].rowItems[0].attributes["data-qa"] == "calculated-summary-answer"
```

Reply:
Whole context is tested now, thanks👍